tests/unit/test_ad.py

```diff
@@ -521,6 +520,174 @@ def test_variable_combinations(grids, variables):
+        _, b0 = eq.assemble()
+
+        return (-1) * np.copy(b0)
+
+    # Setting equations with approximating operators
+    Z = reset_values(gbs, eq_manager)
+
+    line_a = operator_cls(
+        **constructor_kwargs[0], func=line, name="line_a", is_vector_func=False
+    )(x, y)
+
+    circle_a_ = operator_cls(
+        **constructor_kwargs[1], func=circle_, name="circle_a", is_vector_func=False
+    )(x, y)
+    circle_a = circle_a_ - radii**2
+
+    eq_manager.equations = {"line": line_a, "circle": circle_a}
+    iterations = "NOT CONVERGED"
+    res_norm = None
+
+    for i in range(4000):
+
+        A, b = eq_manager.assemble()
+
+        res_norm = np.linalg.norm(b)
+        if res_norm <= 1.0e-10:
+            iterations = i
+            break
+
+        dz = sps.linalg.spsolve(A, b)
+
+        Z += dz
+        dof_manager.distribute_variable(Z)
+        dof_manager.distribute_variable(Z, to_iterate=True)
+
+    assert iterations != "NOT CONVERGED"
+
+    ############ mD domain for test of dimensions
+    # this grid has in total 200 cell DOFs
+    # 88 cells in the 2D domain,
+    # 4 cells in the 1D domain,
+    # 8x2 cells on the Mortar grid
+    coord_point = np.array([[0.2, 0.8], [0.5, 0.5]])
+    indices_point = np.array([[0], [1]])
+    domain = {"xmin": 0, "xmax": 1, "ymin": 0, "ymax": 1}
+    fracture_network = pp.FractureNetwork2d(coord_point, indices_point, domain)
+    mesh_args = {"mesh_size_frac": 0.3}
+    gbmd = fracture_network.mesh(mesh_args)
+    gbmd.compute_geometry()
+
+    # creating variables
+    for g, d in gbmd:
+        d[pp.STATE] = {}
+        d[pp.STATE][pp.ITERATE] = {}
+        d[pp.PRIMARY_VARIABLES] = {}
+
+        d[pp.PRIMARY_VARIABLES].update({"x": {"cells": 1}})
+        d[pp.PRIMARY_VARIABLES].update({"y": {"cells": 1}})
+
+        d[pp.STATE]["x"] = np.ones(g.num_cells)
+        d[pp.STATE][pp.ITERATE]["x"] = np.ones(g.num_cells)
+        d[pp.STATE]["y"] = np.ones(g.num_cells)
+        d[pp.STATE][pp.ITERATE]["y"] = np.ones(g.num_cells)
+
+    # Then for the edges
+    for _, d in gbmd.edges():
+        if d["mortar_grid"].codim == 2:
+            continue
+        else:
+            d[pp.PRIMARY_VARIABLES] = {"mortar_x": {"cells": 1}}
+            d[pp.PRIMARY_VARIABLES].update({"mortar_y": {"cells": 1}})
+
+    dof_manager = pp.DofManager(gbmd)
+    eq_manager = pp.ad.EquationManager(gbmd, dof_manager)
+
+    x = eq_manager.merge_variables([(g, "x") for g, _ in gbmd])
+    y = eq_manager.merge_variables([(g, "y") for g, _ in gbmd])
+    # mortar_x = eq_manager.merge_variables([(e, "mortar_x") for e, _ in gbmd.edges()])
+    # mortar_y = eq_manager.merge_variables([(e, "mortar_y") for e, _ in gbmd.edges()])
+
+    initial_values = np.ones(dof_manager.num_dofs())
+    dof_manager.distribute_variable(initial_values)
+
+    multiply = lambda x, y: x * y
+    op = operator_cls(**constructor_kwargs[0], func=multiply, name="bb-multip")(x, y)
```

Comment:
The problem with the general operator functions is, that the validity of the analytical implementation for the Jac in `Ad_array` is overwritten with something arbitrary (from a programming point of view).
Any non-exact function can return anything as the Jacobian, leading to errors somewhere else in the whole framework.
So the motivation of this test is to avoid exactly that, since we know how the AD framework with operators work and how the Jacobians are supposed to be combined in the end.